modules/issue_tracker/jsx/issueIndex.js

```diff
@@ -474,24 +474,82 @@ class CollapsibleComment extends React.Component {
       "Show Comment History" :
       "Hide Comment History");
 
+    const hrStyle = {
+      marginTop: "8px",
+      marginBottom: "8px"
+    };
+    const colStyle = {
+      paddingRight: "8px"
+    };
     const commentHistory = this.props.commentHistory;
+    let curDateAddedF = null;
+    let curDateAdded = null;
+    let curAddedBy = null;
+    let curState = null;
+    let fieldUsed = null;
+    let fieldChangeArr = null;
+    commentHistory.push({
+      fieldChanged: "SENTINAL",
+      isSentinal: true
+    });
+
     for (let commentID in commentHistory) {
-      if (commentHistory.hasOwnProperty(commentID)) {
-        let action = " updated the " + commentHistory[commentID].fieldChanged + " to ";
-        if (commentHistory[commentID].fieldChanged === 'comment') {
-          action = " commented ";
+      if (!commentHistory.hasOwnProperty(commentID)) {
+        continue;
+      }
+      let change = commentHistory[commentID];
+      let nxtState = (change.fieldChanged === "comment") ?
+        "commented" : "updated";
+      if (curDateAdded !== change.dateAdded || curAddedBy !== change.addedBy || change.isSentinal) {
+        if (fieldChangeArr !== null) {
+          fieldChangeArr.reverse();
+          historyText.push(
+            <div key={"comment_" + commentID}>
```

Comment:
I'm not sure what most of this code does, but putting a single comment in its own `<table>` contained within a `<div>` does not look right